### ISSUE TITLE
failureRedirect is not followed if it fails during OAuth dance

### ISSUE THREAD

**Issue by ekryski** [feathersjs/feathers-authentication-oauth1#8]
### Steps to reproduce

When using with oauth1 plugin the `failureRedirect` option is never used by `auth.express.authenticate()` when it receives callback with error e.g. `/auth/twitter/callback?error=access_denied&error_code=200&error_description=Permissions+error&error_reason=user_denied#_=_` and it will throw NotAuthenticated error instead.

### Expected behavior
When callback URL with error is triggered, and failureRedirect is provided, it should redirect accordingly.

### Actual behavior
NotAuthenticated error is thrown.

### System configuration

**Module versions** (especially the part that's not working):
```
+-- feathers@2.0.3
+-- feathers-authentication@1.0.2
+-- feathers-authentication-jwt@0.3.1
+-- feathers-authentication-oauth1@0.2.3
```

**NodeJS version**: 6.9.1

**Operating System**: Windows 10

**Browser Version**: Chrome 55.0.2883.87 m

**Module Loader**: Node.js

**Comment by ekryski** [comment]
Related to https://github.com/feathersjs/feathers-authentication/issues/387

**Comment by ekryski** [comment]
Closed in v0.2.4
